src/components/BetterDrawerLayout.tsx

```diff
@@ -0,0 +1,611 @@
+      } else {
+        // refer to the large comment above
+        if (
+          drawerType === 'front' &&
+          event.translationX > 0 &&
+          drawerOffset.value < 0
+        ) {
+          newOffset += dragStartPosition.value - (containerWidth - drawerWidth);
+        }
+
+        // clamp the offset so the drawer does not move away from the edge
+        newOffset = Math.max(-drawerWidth, Math.min(0, newOffset));
+      }
+
+      drawerOffset.value = newOffset;
+
+      // send event if there is a listener
+      if (props.onDrawerSlide !== undefined) {
+        runOnJS(props.onDrawerSlide)(openingProgress.value);
+      }
+    } else {
+      // drawerState is SETTLING, save the translation to ignore it later
+      ignoredOffset.value = event.translationX;
+    }
+  });
+  pan.onEnd((_event) => {
+    'worklet';
+    if (drawerState.value === DRAGGING) {
+      // update offsets and animations only when the drag was not ignored
+      drawerSavedOffset.value = drawerOffset.value;
+
+      // if the drawer was dragged more than half of its width open it,
+      // otherwise close it
+      if (fromLeft) {
+        if (drawerOffset.value > drawerWidth / 2) {
+          open();
+        } else {
+          close();
+        }
+      } else {
+        if (drawerOffset.value < -drawerWidth / 2) {
+          open();
+        } else {
+          close();
+        }
+      }
+    }
+  });
+
+  const dynamicDrawerStyles = {
+    backgroundColor: props.drawerBackgroundColor,
+    width: drawerWidth,
+  };
+
+  const drawerStyle = useAnimatedStyle(() => {
+    let translation = 0;
+
+    if (drawerSlide) {
+      // drawer is supposed to be moved with the gesture (in this case
+      // drawer is anchored to be off the screen when not opened)
+      if (fromLeft) {
+        translation = -drawerWidth;
+      } else {
+        translation = containerWidth;
+      }
+      translation += drawerOffset.value;
+    } else {
+      // drawer is stationary (in this case drawer is below the content
+      // so it's anchored left edge to left edge or right to right)
+      if (fromLeft) {
+        translation = 0;
+      } else {
+        translation = containerWidth - drawerWidth;
+      }
+    }
+
+    // if the drawer is not visible move it off the screen to prevent it
+    // from intercepting touch events on Android
+    if (drawerOffset.value === 0) {
+      translation = 10000;
+    }
+
+    return {
+      flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+      transform: [{ translateX: translation }],
+    };
+  }, [drawerSlide, drawerWidth, containerWidth, fromLeft]);
+
+  const containerStyle = useAnimatedStyle(() => {
+    let translation = 0;
+
+    if (containerSlide) {
+      // the container should be moved with the gesture
+      translation = drawerOffset.value;
+    }
+
+    return {
+      transform: [{ translateX: translation }],
+    };
+  }, [containerSlide]);
+
+  if (ref != null) {
```

Comment:
```suggestion
  if (ref !== null) {
```

Reply:
If its type is `null | undefined` then leave it.

Reply:
Changed it in 4c05343f1bc75bb5a3019e20421322641a5f754e